sbol_utilities/default_components.py

```diff
@@ -0,0 +1,148 @@
+from typing import List, Tuple
+
+import sbol3
+import tyto
+
+def promoter(name: str, description: str, sequence: str)-> Tuple[sbol3.Component, sbol3.Sequence]:
+    """
+    Creates a promoter and a sequence Components with a given name, description, and sequence.
+    """
+    promoter_seq = sbol3.Sequence(f'{name}_seq')
+    promoter_seq.elements= sequence
+    promoter_seq.encoding = 'https://identifiers.org/edam:format_1207'
+
+    promoter = sbol3.Component(name, sbol3.SBO_DNA)
+    promoter.name = name
+    promoter.description = description
+    promoter.roles. append(sbol3.SO_PROMOTER)
+    promoter.sequences.append(promoter_seq)
+    return tuple([promoter, promoter_seq])
+
+def rbs(name: str, description: str, sequence: str)-> Tuple[sbol3.Component, sbol3.Sequence]:
+    """
+    Creates a Ribosome Binding Site (RBS) and a sequence Components with a given name, description, and sequence.
+    """
+    rbs_seq = sbol3.Sequence(f'{name}_seq')
+    rbs_seq.elements= sequence
+    rbs_seq.encoding = 'https://identifiers.org/edam:format_1207'
+
+    rbs = sbol3.Component(name, sbol3.SBO_DNA)
+    rbs.name = name
+    rbs.description = description
+    rbs.roles. append(sbol3.SO_RBS)
+    rbs.sequences.append(rbs_seq)
+    return tuple([rbs, rbs_seq])
+
+def cds(name: str, description: str, sequence: str)-> Tuple[sbol3.Component, sbol3.Sequence]:
+    """
+    Creates a coding sequence (CDS) and a sequence Components with a given name, description, and sequence.
+    """
+    cds_seq = sbol3.Sequence(f'{name}_seq')
+    cds_seq.elements= sequence
+    cds_seq.encoding = 'https://identifiers.org/edam:format_1207'
+
+    cds = sbol3.Component(name, sbol3.SBO_DNA)
+    cds.name = name
+    cds.description = description
+    cds.roles. append(sbol3.SO_CDS)
+    cds.sequences.append(cds_seq)
+    return tuple([cds, cds_seq])
+
+def terminator(name: str, description: str, sequence: str)-> Tuple[sbol3.Component, sbol3.Sequence]:
+    """
+    Creates a terminator and a sequence Components with a given name, description, and sequence.
+    """
+    terminator_seq = sbol3.Sequence(f'{name}_seq')
+    terminator_seq.elements= sequence
+    terminator_seq.encoding = 'https://identifiers.org/edam:format_1207'
+
+    terminator = sbol3.Component(name, sbol3.SBO_DNA)
+    terminator.name = name
+    terminator.description = description
+    terminator.roles. append(sbol3.SO_TERMINATOR)
+    terminator.sequences.append(terminator_seq)
+    return tuple([terminator, terminator_seq])
+
+def deg_tag(name: str, description: str, sequence: str)-> Tuple[sbol3.Component, sbol3.Sequence]:
```

Comment:
It's odd to have this one be abbreviated while all the others match their SO names.

Reply:
Agree if they have a name it should be the SO names I will change this function name for protein_stability_element.